engine/order_manager.go

```diff
@@ -284,6 +284,49 @@ func (m *OrderManager) validate(newOrder *order.Submit) error {
 	return nil
 }
 
+// Modify depends on the order.Modify.ID and order.Modify.Exchange fields to uniquely
+// identify an order to modify.
+func (m *OrderManager) Modify(mod *order.Modify) (*order.ModifyResponse, error) {
+	if m == nil {
+		return nil, fmt.Errorf("order manager %w", ErrNilSubsystem)
+	}
+	if atomic.LoadInt32(&m.started) == 0 {
+		return nil, fmt.Errorf("order manager %w", ErrSubSystemNotStarted)
+	}
+
+	exch := m.orderStore.exchangeManager.GetExchangeByName(mod.Exchange)
+	if exch == nil {
+		return nil, ErrExchangeNotFound
+	}
+
+	id, err := exch.ModifyOrder(mod)
```

Comment:
I think you should be updating the ModifyOrder function to return an `order.Modify` that is populated by supported fields. eg in ftx_wrapper, we know that the price and size is the only fields that can be modifed for an order. If the submitted Modify request is successful, we only pass back that amount and price. This will satisfy your issue in regards to unsupported data being populated. It needs to be at the exchange/exchange wrapper level to know what fields it can actually pass back.

Furthermore, you can validate requests and the exchange wrapper level. Again, knowing that FTX can only modify price and amount, if other fields are populated in the request, return an error. We don't want the situation where someone submits a modify request to make the asset type "Futures", the response is successful, but the asset type is unchanged because its unsupported.